src/main/kotlin/org/rust/ide/docs/RustDocumentationProvider.kt

```diff
@@ -60,15 +60,32 @@ class RustDocumentationProvider : AbstractDocumentationProvider() {
         }
 
         val signatureStart = signatureStartElement?.startOffsetInParent ?: 0
-        val signatureEnd = block?.startOffsetInParent ?: textLength
+
+        // pick last of: where clause, return type, or closing brace of the parameters
+        // if all else fails, drop down to the length of the current element
+        val signatureEnd = if (block != null)
+            block!!.startOffsetInParent
```

Comment:
`block` is the body of the function inside (and including) `{` and `}`. So the `startOffsetInParent` is the offset `{`. 

We probably don't need this case, as we now explicitly handle all three possible endings of the function signature.